pkg/brokerapi/openservicebroker/open_service_broker_client.go

```diff
@@ -78,59 +82,79 @@ func (c *openServiceBrokerClient) GetCatalog() (*brokerapi.Catalog, error) {
 	}
 	defer resp.Body.Close()
 
-	// TODO: Align this with the actual response model.
-	si := brokerapi.ServiceInstance{}
-	if err := util.ResponseBodyToObject(resp, &si); err != nil {
-		return nil, err
+	createServiceInstanceResponse := brokerapi.CreateServiceInstanceResponse{}
+	if err := util.ResponseBodyToObject(resp, &createServiceInstanceResponse); err != nil {
+		return nil, fmt.Errorf("Failed to parse broker response: %s", err.Error())
+	}
+
+	switch resp.StatusCode {
+	case http.StatusCreated:
+		return &createServiceInstanceResponse, nil
+	case http.StatusOK:
+		return &createServiceInstanceResponse, nil
+	case http.StatusAccepted:
+		glog.Infof("Asynchronous response received. Polling broker.")
+		err := c.pollBroker(ID, createServiceInstanceResponse.Operation)
+		if err != nil {
+			return nil, err
+		}
+
+		return &createServiceInstanceResponse, nil
+	case http.StatusConflict:
+		return nil, errors.New("Service instance with same id but different attributes exists")
+	case http.StatusUnprocessableEntity:
+		return nil, errors.New("Broker only supports asynchronous provisioning")
+	default:
+		return nil, fmt.Errorf("Unexpected status code %v from broker response", resp.StatusCode)
 	}
-	return &si, nil
 }
 
-func (c *openServiceBrokerClient) UpdateServiceInstance(ID string, req *brokerapi.ServiceInstanceRequest) (*brokerapi.ServiceInstance, error) {
+func (c *openServiceBrokerClient) UpdateServiceInstance(ID string, req *brokerapi.CreateServiceInstanceRequest) (*brokerapi.ServiceInstance, error) {
 	// TODO: https://github.com/kubernetes-incubator/service-catalog/issues/114
 	return nil, fmt.Errorf("Not implemented")
 }
 
-func (c *openServiceBrokerClient) DeleteServiceInstance(ID string) error {
+func (c *openServiceBrokerClient) DeleteServiceInstance(ID string, req *brokerapi.DeleteServiceInstanceRequest) error {
 	url := fmt.Sprintf(serviceInstanceFormatString, c.broker.Spec.URL, ID)
-
 	// TODO: Handle the auth
-	deleteHTTPReq, err := http.NewRequest("DELETE", url, nil)
+	resp, err := util.SendRequest(c.client, http.MethodDelete, url, req)
 	if err != nil {
-		glog.Errorf("Failed to create new HTTP request: %v", err)
-		return err
-	}
-
-	glog.Infof("Doing a request to: %s", url)
-	resp, err := c.client.Do(deleteHTTPReq)
-	if err != nil {
-		glog.Errorf("Failed to DELETE: %#v", err)
-		return err
+		return fmt.Errorf("Failed to send request: %s", err.Error())
 	}
 	defer resp.Body.Close()
 
-	return nil
+	deleteServiceInstanceResponse := brokerapi.DeleteServiceInstanceResponse{}
+	if err := util.ResponseBodyToObject(resp, &deleteServiceInstanceResponse); err != nil {
+		return fmt.Errorf("Failed to parse broker response: %s", err.Error())
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		return nil
+	case http.StatusAccepted:
+		glog.Infof("Asynchronous response received. Polling broker.")
+		err := c.pollBroker(ID, deleteServiceInstanceResponse.Operation)
```

Comment:
fwiw, I'm ok with that for now. we can put in an issue to make that better in the near future